static/js/popovers.js

```diff
@@ -479,14 +493,26 @@ exports.toggle_actions_popover = function (element, id) {
         };
 
         var ypos = elt.offset().top;
-        elt.popover({
-            // Popover height with 7 items in it is ~190 px
-            placement: message_viewport.height() - ypos < 220 ? 'top' : 'bottom',
-            title: "",
-            content: render_actions_popover_content(args),
-            html: true,
-            trigger: "manual",
-        });
+        if (message_viewport.width() > 320) {
+            elt.popover({
+                // Popover height with 7 items in it is ~190 px
+                placement: message_viewport.height() - ypos < 220 ? 'top' : 'bottom',
+                title: "",
+                content: render_actions_popover_content(args),
+                html: true,
+                trigger: "manual",
+            });
+        } else {
+            elt.popover({
+                // Popover height with 7 items in it is ~190 px
+                placement: message_viewport.height() - ypos < 220 ? 'top' : 'bottom',
+                template: render_no_arrow_popover({class: "message-info-popover"}),
+                title: "",
+                content: render_actions_popover_content(args),
+                html: true,
+                trigger: "manual",
+            });
+        }
```

Comment:
@timabbott is there a way to decrease the duplication of code here?
This workaround works well but my concern is with the duplicate code.

Reply:
Yes, let's add a `
popover_args = {
                    title: "",
￼                content: render_actions_popover_content(args),
￼                html: true,
￼                trigger: "manual",
}
```
and then the two cases can just set a couple values in `popover_args` and then return to common code that calls `elt.popover`.

Reply:
It seems that it cannot be declared directly. Do I need to use JSON.parse(text) here ?

Reply:
Declaring it directly gives errors
![Screenshot from 2019-10-15 04-05-18](https://user-images.githubusercontent.com/41019632/66787377-194d1a80-ef01-11e9-92a3-73a0a4b13e24.png)

Reply:
ok, so I realized that the problem lies with
```
content : render_action_popover_content(args),
```
as it has `args` as an argument

Reply:
On assessing all this, I believe a bit of extra code to simplify things is better than making it shorter by complicating it and declaring further new variables.

Reply:
@timabbott I have successfully removed the duplicate code! Had to do some digging through Bootstrap docs. Please review now.

Reply:
why is it failing a backend test?

```
----
Running ./tools/test-backend --include-webhooks
Traceback (most recent call last):
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/srv/zulip/zerver/lib/db.py", line 31, in execute
    return wrapper_execute(self, super().execute, query, vars)
  File "/srv/zulip/zerver/lib/db.py", line 18, in wrapper_execute
    return action(sql, params)
psycopg2.errors.UndefinedColumn: column zerver_realm.user_group_edit_policy does not exist
LINE 1: ...olicy", "zerver_realm"."invite_to_stream_policy", "zerver_re...
                                                             ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "./tools/test-backend", line 472, in <module>
    main()
  File "./tools/test-backend", line 378, in main
    TestRunner = get_runner(settings)
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/test/utils.py", line 325, in get_runner
    test_module = __import__(test_module_name, {}, {}, force_str(test_path[-1]))
  File "/srv/zulip/zerver/lib/test_runner.py", line 17, in <module>
    from zerver.lib import test_classes, test_helpers
  File "/srv/zulip/zerver/lib/test_classes.py", line 100, in <module>
    class ZulipTestCase(TestCase):
  File "/srv/zulip/zerver/lib/test_classes.py", line 131, in ZulipTestCase
    DEFAULT_REALM = Realm.objects.get(string_id='zulip')
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/db/models/query.py", line 374, in get
    num = len(clone)
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/db/models/query.py", line 232, in __len__
    self._fetch_all()
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/db/models/query.py", line 1121, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/db/models/query.py", line 53, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch)
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/db/models/sql/compiler.py", line 899, in execute_sql
    raise original_exception
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/db/models/sql/compiler.py", line 889, in execute_sql
    cursor.execute(sql, params)
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 79, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/utils/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/srv/zulip/zerver/lib/db.py", line 31, in execute
    return wrapper_execute(self, super().execute, query, vars)
  File "/srv/zulip/zerver/lib/db.py", line 18, in wrapper_execute
    return action(sql, params)
django.db.utils.ProgrammingError: column zerver_realm.user_group_edit_policy does not exist
LINE 1: ...olicy", "zerver_realm"."invite_to_stream_policy", "zerver_re...
                                                             ^


FAILED ./tools/test-backend --include-webhooks

```